Signal/src/contact/OWSContactsManager.m

```diff
@@ -399,12 +399,26 @@ - (BOOL)phoneNumber:(PhoneNumber *)phoneNumber1 matchesNumber:(PhoneNumber *)pho
 
 #pragma mark - Whisper User Management
 
+- (BOOL)hasNameInSystemContactsForRecipientId:(NSString *)recipientId
+{
+    return [self cachedDisplayNameForRecipientId:recipientId] != nil;
+}
+
 - (NSString *)unknownContactName
 {
     return NSLocalizedString(@"UNKNOWN_CONTACT_NAME",
                              @"Displayed if for some reason we can't determine a contacts phone number *or* name");
 }
 
+- (nullable NSString *)formattedProfileNameForRecipientId:(NSString *)recipientId
+{
+    NSString *_Nullable profileName = [self.profileManager profileNameForRecipientId:recipientId];
+    if (profileName == nil) {
```

Comment:
a) let's make sure we're stripping leading and trailing whitespace from profile names when we decrypt them. I already do so for local user's profile name changes.
b) I'm thinking we should also write and use a function to filter profile names for invalid characters and apply it to incoming and outgoing profile names.  I'm sure we'll add more and more rules to that filter with time, e.g. the unicode character the changes the orientation of text, etc.
c) I'm using `profileName.length > or == 0` instead of nil checks so that we handle empty names appropriately.